src/walker.rs

```diff
@@ -155,6 +155,7 @@ impl Walker {
       file_infos.push(FileInfo {
         path: file_path,
         length: Bytes(len),
+        mtime: metadata.modified().ok(),
```

Comment:
We should return an error if we can't read the modificiation time.

Reply:
I think storing it as an `Option<SystemTime>` is reasonable. On comparison I would do:

```
a.mtime.context(error::ModificationTimeMissing)?
  .cmp(b.mtime.context(error::ModificationTimeMissing)?)
````

with a new error variant, `ModificationTimeMissing`. So the error is only returned if the modification time is missing and the user tries to sort on it.

Reply:
The standard slices' `sort_by` requires the comparator to return `Ordering`, so it seems that panicking is the only way to express an error there.